pkg/daemon/controller/controller.go

```diff
@@ -276,83 +284,6 @@ func (sc *ServiceController) Reconcile(_ context.Context, req ctrl.Request) (res
 	return ctrl.Result{}, nil
 }
 
-// reconcileEnvironmentVariables makes sure that the proxy variables exist as expected, or are safely rectified.
-// Returns a boolean expressing whether the instance is awaiting a reboot.
-func (sc *ServiceController) reconcileEnvironmentVariables(envVars map[string]string, node core.Node) (bool, error) {
-	restartRequired, err := sc.ensureVarsAreUpToDate(envVars)
-	if err != nil {
-		return false, err
-	}
-	if restartRequired {
-		// Applying the reboot annotation results in an event picked up by WMCO's node controller to reboot the instance
-		if err = metadata.ApplyRebootAnnotation(sc.ctx, sc.client, node); err != nil {
-			return false, fmt.Errorf("error setting reboot annotation on node %s: %w", sc.nodeName, err)
-		}
-		return true, nil
-	}
-	// Wait until the environment variables at the process level are set as expected.
-	// This will only be picked up after WMCO reboots the instance
-	err = wait.PollUntilContextTimeout(sc.ctx, 15*time.Second, 5*time.Minute, true,
-		func(ctx context.Context) (done bool, err error) {
-			stillNeedsReboot := false
-			for _, varName := range cluster.SupportedProxyVars {
-				cmd := fmt.Sprintf("[Environment]::GetEnvironmentVariable('%s', 'Process')", varName)
```

Comment:
I always felt like this check was an overreach by WICD. What we were doing with this is ensuring Windows OS is working as expected i.e. that when we reboot, the OS propagates system level env vars to services. We should only be waiting for a reboot

Reply:
> What we were doing with this is ensuring Windows OS is working as expected

Not really, WICD needs to check the current state of the VM and respond accordingly. What if WICD crashed right after changing the env variables and failed to apply the annotation? we can't assume a restart happened just because WMCO starts and there is no restart annotation.